src/Moryx.Notifications/Adapter/NotificationAdapter.cs

```diff
@@ -222,17 +228,25 @@ void INotificationSourceAdapter.PublishProcessed(INotification notification)
                 _pendingPubs.Remove(map);
                 _published.Add(map);
                 _listLock.ExitWriteLock();
+                return;
             }
-            else
-            {
-                // Notification is maybe not pending anymore - we only can acknowledge it
-                _listLock.ExitWriteLock();
 
-                var managed = (IManagedNotification)notification;
+            // Notification is maybe not pending anymore
+            _listLock.ExitWriteLock();
+            var managed = (IManagedNotification)notification;
+
+            // If necessary we can acknowledge it
+            if (managed.Acknowledged is null)
+            {
+                Logger?.Log(LogLevel.Error, "Notification was removed from the pending publications " +
```

Comment:
Are these null checks really necessary? Normally the dependencies are not checked.